samples/ChangefeedMigrationSample/Program.cs

```diff
@@ -0,0 +1,193 @@
+﻿namespace ChangefeedMigrationSample
+{
+    using DocumentDB.ChangeFeedProcessor;
+    using Microsoft.Azure.Documents;
+    using Microsoft.Azure.Documents.Client;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    // ------------------------------------------------------------------------------------------------
+    // This sample demonstrates using change processor library to read changes from source collection 
+    // to destination collection 
+    // ------------------------------------------------------------------------------------------------
+
+    public class DeviceReading
+    {
+        [JsonProperty("id")]
+        public string Id;
+
+        [JsonProperty("deviceId")]
+        public string DeviceId;
+
+        [JsonConverter(typeof(IsoDateTimeConverter))]
+        [JsonProperty("readingTime")]
+        public DateTime ReadingTime;
+
+        [JsonProperty("metricType")]
+        public string MetricType;
+
+        [JsonProperty("unit")]
+        public string Unit;
+
+        [JsonProperty("metricValue")]
+        public double MetricValue;
+    }
+
+    class Program
+    {
+        private const string EndPointUrl = "https://interntest.documents.azure.com:443/";
+        private const string PrimaryKey = "mXBmwssUDqDNL03M0qkmMBYizwpeIrLqCyFNUOGQsGCEeLRRkWJDleEORnVNzfQ13dkiIyxfhgVM4QAQLzQQzg==";
+        private const string DBName = "SmartHome";
+        private const string MonitoredCollectionName = "Nest";
+        private const string LeaseCollectionName = "Lease";
+        private const string DestCollName = "Dest";
+        private const string DestCollPartKey = "/id";
+
+        static void Main(string[] args)
+        {
+            Console.WriteLine("ChangeFeed App");
+            Program newApp = new ChangefeedMigrationSample.Program();
+            // Thread 1 comment out for thread 2 
+            newApp.RunChangeFeedProcessor(EndPointUrl, PrimaryKey);
+            // Thread 2 comment out for thread 1 
+            // UpdateDb(EndPointUrl, PrimaryKey); 
+            Console.ReadKey();
+        }
+
+        async void RunChangeFeedProcessor(string uri, string secretKey)
+        {
+            // connect client 
+            DocumentClient client = new DocumentClient(new Uri(uri), secretKey);
+            await client.CreateDatabaseIfNotExistsAsync(new Database { Id = DBName });
+
+            // create monitor collection if it does not exist
+            await client.CreateDocumentCollectionIfNotExistsAsync(
+                UriFactory.CreateDatabaseUri(DBName),
+                new DocumentCollection { Id = MonitoredCollectionName },
+                new RequestOptions { OfferThroughput = 2500 });
```

Comment:
As a matter of style we like to put "this" in front of all class variables.  So in this case it would be this.MonitoredCollectionName